src/cljs/main/broadfcui/common/management_utils.cljs

```diff
@@ -0,0 +1,134 @@
+(ns broadfcui.common.management-utils
+  (:require
+    [dmohs.react :as react]
+    [broadfcui.common :as common]
+    [broadfcui.common.components :as comps]
+    [broadfcui.common.flex-utils :as flex]
+    [broadfcui.common.input :as input]
+    [broadfcui.common.modal :as modal]
+    [broadfcui.common.style :as style]
+    [broadfcui.common.table.table :as table]
+    [broadfcui.common.table.style :as table-style]
+    [broadfcui.endpoints :as endpoints]
+    [broadfcui.utils :as utils]
+    ))
+
+(react/defc AddUserDialog
+  {:render
+   (fn [{:keys [props state this refs]}]
+     [comps/OKCancelForm
+      {:header (str "Add user to " (:group-name props))
+       :ok-button #(this :add-user)
+       :get-first-element-dom-node #(react/find-dom-node (@refs "email"))
+       :content
+       (react/create-element
+        [:div {:style {:width 420}}
+         (when (:adding? @state)
+           [comps/Blocker {:banner "Adding user..."}])
+         [:div {:style {:display "flex"}}
+          [:div {:style {:flex "1 1 auto"}}
+           (style/create-form-label "User email")
+           [input/TextField {:ref "email" :style {:width "100%"}
+                             :predicates [(input/valid-email "Email")]
+                             :onKeyDown (common/create-key-handler [:enter] #(this :add-user))}]]
+          [:div {:style {:flex "0 0 10px"}}]
+          [:div {:style {:flex "0 0 100px"}}
+           (style/create-form-label "Role")
+           (style/create-identity-select {:ref "role"} ["User" "Owner"])]]
+         (:footer props)
+         (style/create-validation-error-message (:fails @state))
+         [comps/ErrorViewer {:error (:server-error @state)
+                             :expect {404 "This is not a registered user"}}]])}])
+   :add-user
```

Comment:
Since you're touching this, perhaps `:-add-user` to continue with the convention for private methods.